### PR TITLE
[lldb-dap] Fixing a type encoding issue with dap Stopped events.

### PR DESCRIPTION
**Pull request by ashgti**
Previously the type of the breakpoint id in the Stopped event was a uint64_t, however thats the wrong type for a breakpoint id, which can cause encoding issues when internal breakpoints are hit.

### CODE REVIEW

File: lldb/tools/lldb-dap/JSONUtils.cpp [hunk 1]
```diff
@@ -953,9 +953,9 @@ llvm::json::Value CreateThreadStopped(lldb::SBThread &thread,
     } else {
       body.try_emplace("reason", "breakpoint");
       char desc_str[64];
-      uint64_t bp_id = thread.GetStopReasonDataAtIndex(0);
-      uint64_t bp_loc_id = thread.GetStopReasonDataAtIndex(1);
-      snprintf(desc_str, sizeof(desc_str), "breakpoint %" PRIu64 ".%" PRIu64,
+      break_id_t bp_id = thread.GetStopReasonDataAtIndex(0);
+      break_id_t bp_loc_id = thread.GetStopReasonDataAtIndex(1);
+      snprintf(desc_str, sizeof(desc_str), "breakpoint %" PRIo32 ".%" PRIo32,
```
Comment (clayborg) [inline lldb/tools/lldb-dap/JSONUtils.cpp:958]: This is indeed octal which isn't what we want. You need to use `PRIi32`

**Review by clayborg — APPROVED** [review]

File: lldb/tools/lldb-dap/JSONUtils.cpp [hunk 2]
```diff
@@ -953,9 +953,9 @@ llvm::json::Value CreateThreadStopped(lldb::SBThread &thread,
     } else {
       body.try_emplace("reason", "breakpoint");
       char desc_str[64];
-      uint64_t bp_id = thread.GetStopReasonDataAtIndex(0);
-      uint64_t bp_loc_id = thread.GetStopReasonDataAtIndex(1);
-      snprintf(desc_str, sizeof(desc_str), "breakpoint %" PRIu64 ".%" PRIu64,
+      break_id_t bp_id = thread.GetStopReasonDataAtIndex(0);
```
Comment (DavidGoldman) [inline lldb/tools/lldb-dap/JSONUtils.cpp:956]: This looks like an implicit cast from uint64_t to int32_t, how safe is this?

Comment (clayborg) [reply lldb/tools/lldb-dap/JSONUtils.cpp:956]: We have an abstract API to access stop reason data via `uint64_t SBThread::GetStopReasonDataAtIndex(...)` which returns a `uint64_t`, but the `break_id_t` is defined as `typedef int32_t break_id_t;`. If we want to show the breakpoint ID correctly as a negative number for internal breakpoints, then this should be ok.

File: lldb/tools/lldb-dap/JSONUtils.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashgti) [reply lldb/tools/lldb-dap/JSONUtils.cpp]: Switched to llvm::vformat instead.

Comment (ashgti) [reply lldb/tools/lldb-dap/JSONUtils.cpp]: Switched to `llvm::vformat`.

File: lldb/tools/lldb-dap/JSONUtils.cpp (hunk 2 above)
Comment (ashgti) [reply lldb/tools/lldb-dap/JSONUtils.cpp]: These are implicitly cast here https://github.com/llvm/llvm-project/blob/e823136d43c40b0a9ba6930fd285768f1b46fcb6/lldb/source/API/SBThread.cpp#L251C35-L251C48 to uint64_t, this is converting it back to the internal type. This should be okay.

**Pull request merged**